ascon-aead/Cargo.toml

```diff
@@ -0,0 +1,38 @@
+[package]
+name = "ascon-aead"
+version = "0.4.2-pre"
+description = "Implementation of the authenticated encryption schemes Ascon-128, Ascon-128a, and Ascon-80pq"
+authors = [
+    "Sebastian Ramacher <sebastian.ramacher@ait.ac.at>",
+    "RustCrypto Developers",
+]
+edition = "2021"
+license = "Apache-2.0 OR MIT"
+readme = "README.md"
+repository = "https://github.com/RustCrypto/AEADs"
+keywords = ["AEAD", "ascon", "encryption"]
+categories = ["cryptography", "no-std"]
+rust-version = "1.56"
+
+[dependencies]
+aead = { version = "0.5", default-features = false }
+subtle = { version = "2", default-features = false }
+zeroize = { version = "1.5", optional = true, default-features = false, features = [
+    "derive",
+] }
+ascon = "0.3"
+
+[dev-dependencies]
+hex = "0.4"
```

Comment:
FWIW we usually use `hex-literal` for this (which parses hex at compile time and emits `[u8; N]` arrays), but I'd be happy to switch to that in a followup PR

Reply:
The test vectors are currently read during runtime from the data files. A switch to `hex-literal` would require a rewrite of the current tests.

Reply:
@newpavlov `hex-literal` feature request: `include_hex!` which works like `include_bytes!`/`include_str!`

Reply:
@tarcieri
Could you create an issue for it?

BTW I don't think it would've helped here, since the test vector files contain non-hex data. I think a better approach would be to convert them to the blobby format and use the `aead::new_test!` macro`.

Reply:
In the long run I think it would make more sense to support the format of the test vector files. The format is the same (except for the provided fields) is the same for all current standardization processes run by NIST. The test vectors for all the post-quantum signatures and KEMs look the same.

Reply:
The test vectors get included into published packages, so it's worth to minimize them. Even with gzip, binary encoding produces a smaller result.

Reply:
@sebastinas for what it's worth, we have an extractor for a similar use case: extracting files from Wycheproof, so we can convert to a common format:

https://github.com/RustCrypto/utils/tree/master/wycheproof2blb

@newpavlov the minute size wins for "blobby" files over e.g. git's own usage of gzip goes away the instant we need to modify any of the files and git has to duplicate the entire contents of the file because it treats binary data as opaque git objects.

It also removes the ability to add any kind of context to test vectors, like their provenance.

Reply:
For similar test vector files I was using a quick Python script which generates file with only hex data. it then gets converted by the utility in the `blobby` crate. I may do conversion for this crate myself a bit later.

@tarcieri 
I was talking primarily about size of published packages. Not only they get downloaded much more often than repository gets cloned (outside of CI jobs), but we include them with each release.

In practice, we modify `blb` files quite rarely (I can remember only 2-3 cases). Usually, we add new test vectors as separate files even if we test the same algorithm (e.g. we did so with the Wycheproof vectors).

Provenance is better documented not in test vector files, but in documentation of tests which use those files. It's the first place developers will look into when inspecting crate tests. For example, the test files in this PR do not include any provenance information, so we have to document it separately either way.

Reply:
Any time there's been a test failure in the Wycheproof-generated files it's been a massive pain to debug, because while the original files contain detailed information about the test case in question. I have to manually go from the blobby test failure back to the original test vectors to get that information back, which is quite confusing.

In general having a bespoke, opaque binary format without great tooling to assist working with it is an impediment to every new project contributor.